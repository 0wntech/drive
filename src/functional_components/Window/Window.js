import React from 'react';
import classnames from 'classnames';
import styles from './Window.module.scss';
import PropTypes from 'prop-types';
import X from '../../assets/svgIcons/X';
import { ClassicSpinner } from 'react-spinners-kit';

function Window({
    windowName,
    className,
    containerClassName,
    children,
    onClose,
    visible,
    loading,
}) {
    return visible ? (
        <div
            className={styles.wrapper}
            onClick={(e) => {
                if (
                    typeof e.target.className === 'string' &&
                    e.target.className.includes('Window_wrapper')
                )
                    onClose();
            }}
        >
            <div className={classnames(styles.container, containerClassName)}>
                <div className={styles.head}>
                    <span
                        className={styles.title}
                        data-test-id={`window-${windowName.replace(' ', '-')}`}
                    >
                        {windowName}
                    </span>
                    <X
                        className={styles.close}
                        preserveAspectRatio="none"
                        viewBox="5 4 24 24"
                        onClick={onClose}
                    />
                </div>
                <div className={classnames(styles.body, className)}>
                    {loading ? (
                        <ClassicSpinner
                            size={30}
                            color="#686769"
                            loading={loading}
                        />
                    ) : (
                        children
                    )}
                </div>
            </div>
        </div>
    ) : null;
}

Window.propTypes = {
    windowName: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.array,
    onClose: PropTypes.func,
    visible: PropTypes.bool,
};

export default Window;
