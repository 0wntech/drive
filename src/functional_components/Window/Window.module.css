.container {
    z-index: 3;
    position: fixed;
    margin: auto;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    background-color: white;
    min-width: 30%;
    max-width: 30%;
    border-radius: 20px;
}

.opacity {
    z-index: 2;
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: black;
    opacity: 0.5;
}

.close {
    cursor: pointer;
    width: 24px;
    height: 24px;
    justify-self: center;
    background-color: #ff697b;
    border-radius: 100%;
    margin-left: auto;
}

.close:hover {
    transform: scale(1.05);
}

.head {
    height: 50px;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    background-color: #e2e2e6;
    padding: 0px 20px;
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
}

.body {
    padding: 20px;
}

.title {
    font-size: 1.1em;
    font-weight: 300;
    width: 100%;
    position: absolute;
    text-align: center;
}
