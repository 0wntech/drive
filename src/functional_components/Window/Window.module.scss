@import '../../../styles/constants.scss';

.container {
    z-index: 5;
    position: fixed;
    margin: auto;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    background-color: white;
    width: 40vw;
    height: 50vh;
    border-radius: 20px;
    border: 5px solid $dark_grey;
}

.wrapper {
    display: flex;
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    z-index: 4;
    background: rgba($color: #000, $alpha: 0.2);
}

.close {
    cursor: pointer;
    margin-left: auto;
    margin-right: 16px;
    height: 20px;
    width: 20px;
    z-index: 1;
}

.close path {
    fill: #666666;
}

.head {
    height: 50px;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
}

.body {
    padding: 20px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 80%;
}

.title {
    font-size: 1.1em;
    font-weight: 300;
    width: 100%;
    position: absolute;
    text-align: center;
}

@media (max-width: $screen_m) {
    .wrapper {
        align-items: flex-end;
    }

    .container {
        border: none;
        top: initial;
        left: 0;
        transform: initial;
        min-width: initial;
        max-width: initial;
        width: 100%;
        min-height: 50%;
        border-bottom-left-radius: 0;
        border-bottom-right-radius: 0;
    }
}
