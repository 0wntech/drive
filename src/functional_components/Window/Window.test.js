import React from 'react';
import Window from './Window';
import { shallow } from 'enzyme';
import TestRenderer from 'react-test-renderer';

describe('Window', () => {
    it('should render correctly', () => {
        const mockFunction = jest.fn();
        const tree = TestRenderer.create(
            <Window
                windowName="Test Window"
                className="chicken"
                onClose={mockFunction}
                visible
            >
                <div>I'm a child</div>
                <button>child button</button>
            </Window>
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should handle the click event', () => {
        const mockFunction = jest.fn();
        const wrapper = shallow(
            <Window
                windowName="Test Window"
                visible
                className="chicken"
                onClose={mockFunction}
            />
        );
        wrapper.find('.close').simulate('click');
        expect(mockFunction.mock.calls.length).toEqual(1);
    });
});
