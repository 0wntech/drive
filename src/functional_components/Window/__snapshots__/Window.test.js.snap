// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Window should render correctly 1`] = `
<div
  className="wrapper"
  onClick={[Function]}
>
  <div
    className="container"
  >
    <div
      className="head"
    >
      <span
        className="title"
        data-test-id="window-Test-Window"
      >
        Test Window
      </span>
      <svg
        className="close"
        fill="none"
        height={32}
        onClick={[MockFunction]}
        preserveAspectRatio="none"
        viewBox="5 4 24 24"
        width={32}
      >
        <path
          d="M24.933 23.067a1.289 1.289 0 010 1.866c-.266.267-.533.4-.933.4s-.667-.133-.933-.4L16 17.867l-7.067 7.066c-.266.267-.533.4-.933.4s-.667-.133-.933-.4a1.289 1.289 0 010-1.866L14.133 16 7.067 8.933a1.289 1.289 0 010-1.866 1.289 1.289 0 011.866 0L16 14.133l7.067-7.066a1.289 1.289 0 011.866 0 1.289 1.289 0 010 1.866L17.867 16l7.066 7.067z"
          fill="#F41818"
        />
      </svg>
    </div>
    <div
      className="body chicken"
    >
      <div>
        I'm a child
      </div>
      <button>
        child button
      </button>
    </div>
  </div>
</div>
`;
